src/plugins/embeddable/public/lib/actions/edit_panel_action.ts

```diff
@@ -75,13 +75,29 @@ export class EditPanelAction implements Action<ActionContext> {
       embeddable &&
         embeddable.getOutput().editable &&
         (embeddable.getOutput().editUrl ||
-          (embeddable.getOutput().editApp && embeddable.getOutput().editPath))
+          (embeddable.getOutput().editApp && embeddable.getOutput().editPath) ||
+          embeddable.getOutput().editableWithExplicitInput)
     );
     const inDashboardEditMode = embeddable.getInput().viewMode === ViewMode.EDIT;
     return Boolean(canEditEmbeddable && inDashboardEditMode);
   }
 
   public async execute(context: ActionContext) {
+    const embeddable = context.embeddable;
+    const { editableWithExplicitInput } = embeddable.getOutput();
```

Comment:
Is there a reason this is exposed via embeddable output rather than as either a method call or as a property on the embeddable definition? Do you expect this to change during the life time of individual embeddables?

It looks like this matches the current `editable` behaviour, I wonder if we have any instances where this changes.

Reply:
I am not sure! As you said, I followed the `editable` and other `edit*` related props pattern. 

Seems like `editable` can be dynamic, for example: 

```
  private getIsEditable() {
    return (
      this.deps.capabilities.canSaveVisualizations ||
      (!this.inputIsRefType(this.getInput()) && this.deps.capabilities.canSaveDashboards)
    );
  }
```

`editUrl` is also can be dynamic depending on savedObjectId, for example. 

`editApp` don't think so. and `editableWithExplicitInput` also not at least at the moment. 
But I think makes sense to keep these properties together.

Reply:
Sounds good for the moment! Eventually we're hoping to remove the extra step of accessing these sorts of things via output, and standardize on calling them as methods on the embeddable instance, like `getIsEditable`. But that process doesn't have to start quite yet!